### PR TITLE
feat(website): rate pages

### PR DESCRIPTION
**Pull request by TheSisb**
This PR edits the button over the Table of Contents on component pages to ask users to rate our docs. The data is sent to Google Analytics as our backend. It is all hidden inside a neat popover that responds differently based on like/dislike. Users rating is stored in localstorage.

### CODE REVIEW

File: packages/paste-website/src/components/shortcodes/feedback-popover/index.tsx [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  likePage: (event: React.MouseEvent) => void;
+  dislikePage: (event: React.MouseEvent) => void;
+};
+
+const UnratedPage: React.FC<RatingProps> = ({likePage, dislikePage}) => (
+  <>
+    <Heading as="h4" variant="heading40">
+      Rate this page
+    </Heading>
+    <Text as="p" marginBottom="space50">
```
Comment (SiTaggart) [inline packages/paste-website/src/components/shortcodes/feedback-popover/index.tsx:26]: Is it weird that we're not using our Paragraph here?

File: packages/paste-website/src/components/shortcodes/feedback-popover/index.tsx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+      Let us know how we&apos;re doing! Please rate this page:
+    </Text>
+    <Button variant="link" onClick={likePage}>
+      <ThumbsUpIcon size="sizeIcon50" decorative={false} title="Like this page" />
+      It&apos;s useful!
+    </Button>
+    <Box as="span" marginLeft="space60">
```
Comment (SiTaggart) [inline packages/paste-website/src/components/shortcodes/feedback-popover/index.tsx:33]: Not: could use stack 🤷

File: packages/paste-website/src/utils/SimpleStorage.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+try {
+  storage = window.localStorage;
+  // eslint-disable-next-line unicorn/prefer-optional-catch-binding
+} catch (error) {
+  // Ignore
+}
+
+const namespaceKey = (key: string): string => `${BASE_KEY}/${key}`;
+
+// Simple localStorage wrapper w/ namespace support
```
Comment (SiTaggart) [inline packages/paste-website/src/utils/SimpleStorage.ts:21]: Should we start trying to encourage JSDoc style comments? Just as a nice DX it'll give you nice autocomplete information in VS Code.

**Review by SiTaggart — COMMENTED** [review]
Nice and simple 👍

**Comment by katieporter** [comment]
Should the initial focus when opening the popover be on the close button instead, so we aren't unintentionally biasing folks to give a thumbs up (if they accidentally hit enter after opening the popover)?
image

File: packages/paste-website/src/components/shortcodes/feedback-popover/index.tsx [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+);
+
+const DislikedPage: React.FC<RatingProps> = () => (
+  <>
+    <Button variant="link" disabled>
+      <ThumbsDownIcon size="sizeIcon50" decorative={false} title="Dislike this page" />
+      It has issues
+    </Button>
+    <Separator orientation="horizontal" verticalSpacing="space30" />
+    <Text as="p" marginBottom="space20">
```
Comment (katieporter) [inline packages/paste-website/src/components/shortcodes/feedback-popover/index.tsx:65]:
```suggestion
    <Text as="p" marginBottom="space40">
```

File: packages/paste-website/src/components/shortcodes/feedback-popover/index.tsx [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+  </>
+);
+
+const LikedPage: React.FC<RatingProps> = () => (
+  <>
+    <Button variant="link" disabled>
+      <ThumbsUpIcon size="sizeIcon50" decorative={false} title="Like this page" />
+      It&apos;s useful!
+    </Button>
+    <Separator orientation="horizontal" verticalSpacing="space30" />
```
Comment (katieporter) [inline packages/paste-website/src/components/shortcodes/feedback-popover/index.tsx:48]:
```suggestion
    <Separator orientation="horizontal" verticalSpacing="space40" />
```

**Comment by katieporter** [comment]
I really love the simplicity of the choices in the Popover (give a thumbs up, or a thumbs down), but I do wonder if there are some folks who would want to be able to give feedback without giving a thumbs down rating (a thumbs down might seem a little much just because the page has a typo).

I might want to see that "see a mistake?" line come back, if we don't think it complicates things too much.

**Comment by SiTaggart** [comment]
I'm still not entirely sure this is the right icon. Have we considered the support icon? screenshot


<details><summary>Browser metadata</summary>





```
Path:      /components/alert/
Browser:   Edge 92.0.902.55 on Mac OS 10.15.7
Viewport:  3008 x 1556 @2x
Language:  en-AU
Cookies:   Enabled
```

[Open in BrowserStack](https://www.browserstack.com/user/try-live?url=https%3A%2F%2Fdeploy-preview-1669--paste-docs.netlify.app%2Fcomponents%2Falert%2F&os=OS%20X&os_version=Catalina&browser=Edge&browser_version=92.0&resolution=3008x1556&speed=1&start=true&ref=netlify-source)



</details>

[Open Deploy Preview](https://deploy-preview-1669--paste-docs.netlify.app/components/alert/) · [Mark as Resolved](https://app.netlify.com/cdp/resolve?deployID=60ff598f51d5220007490e94&commentID=610031873e66ae44453aefd1&resolution=resolved)

**Comment by SiTaggart** [comment]
In dark mode the heading on the "has issues" state has contrast issues. Lets' ignore the popover triangle for now and file a Jira ticket for that as a separate fix screenshot


<details><summary>Browser metadata</summary>





```
Path:      /components/alert/
Browser:   Edge 92.0.902.55 on Mac OS 10.15.7
Viewport:  3008 x 1556 @2x
Language:  en-AU
Cookies:   Enabled
```

[Open in BrowserStack](https://www.browserstack.com/user/try-live?url=https%3A%2F%2Fdeploy-preview-1669--paste-docs.netlify.app%2Fcomponents%2Falert%2F&os=OS%20X&os_version=Catalina&browser=Edge&browser_version=92.0&resolution=3008x1556&speed=1&start=true&ref=netlify-source)



</details>

[Open Deploy Preview](https://deploy-preview-1669--paste-docs.netlify.app/components/alert/) · [Mark as Resolved](https://app.netlify.com/cdp/resolve?deployID=60ff598f51d5220007490e94&commentID=610031eda835de465460bb17&resolution=resolved)

**Comment by SiTaggart** [comment]
Oh actually even in the success state and light mode the heading is super light. screenshot


<details><summary>Browser metadata</summary>





```
Path:      /components/alert-dialog
Browser:   Edge 92.0.902.55 on Mac OS 10.15.7
Viewport:  3008 x 1556 @2x
Language:  en-AU
Cookies:   Enabled
```

[Open in BrowserStack](https://www.browserstack.com/user/try-live?url=https%3A%2F%2Fdeploy-preview-1669--paste-docs.netlify.app%2Fcomponents%2Falert-dialog&os=OS%20X&os_version=Catalina&browser=Edge&browser_version=92.0&resolution=3008x1556&speed=1&start=true&ref=netlify-source)



</details>

[Open Deploy Preview](https://deploy-preview-1669--paste-docs.netlify.app/components/alert-dialog) · [Mark as Resolved](https://app.netlify.com/cdp/resolve?deployID=60ff598f51d5220007490e94&commentID=610032625053124a87d29e72&resolution=resolved)

File: packages/paste-website/src/components/shortcodes/feedback-popover/index.tsx (hunk 1 above)
Comment (TheSisb) [reply packages/paste-website/src/components/shortcodes/feedback-popover/index.tsx:26]: The Paragraph component either has space70 or space0. For the popover, space70 looked a little too airy, as it competed with the outer padding of the whole popover. That's why I opted for this smaller margin.

**Comment by TheSisb** [comment]
@SiTaggart I made a ticket for the color contrast issue https://issues.corp.twilio.com/browse/DSYS-2954

**Comment by katieporter** [comment]
> @SiTaggart I made a ticket for the color contrast issue https://issues.corp.twilio.com/browse/DSYS-2954

Disabled buttons don't need to pass 3:1, do they?

**Comment by TheSisb** [comment]
> To Simon's point though, a disabled button might not be the best choice here, since it's being used to remind the user what rating they gave, but it's inaccessible to some users. Perhaps just an icon with plain text should be used instead?

Good suggestion, I just made that change.

**Comment by SiTaggart** [comment]
> a disabled button might not be the best choice here

Yeah, I totally didn't get that it was a button. That's what the confusion was

**Comment by gloriliale** [comment]
I think using the term "issue(s)" here in two different places could be an... issue (...I'll see myself out)
It might be better to have the bottom link say "Spot on error?", or to change the language for the thumbs down button to "It needs work".
Screen Shot 2021-08-02 at 12 19 04 PM

**Review by gloriliale — APPROVED** [review]
I really like how this turned out. It's a really pleasant experience to click on this button and be able to rate the page from anywhere! Great work on this v1!

**Review by SiTaggart — APPROVED** [review]

**Pull request merged**